uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequestMailNotification.php

```diff
@@ -0,0 +1,391 @@
+<?php
+
```

Comment:
For the classes what you think about this:
- have an abstract BaseLeaveRequestNotificationTemplate, with all the logic that is shared among the templates
- have a LeaveRequestNotificationTemplate and a TOILRequestNotificationTemplate extending this base abastract class (it will just be an extends, without actually overriding anything)
- have a SicknessRequestNotificatioTemplate which will have the logic for parameters specific for this type

About all the templates having access to the whole LeaveRequest, I think that this is fine. The template idea is just to separate the different logic for each type, but they're all still based on a LeaveRequest.